docs/GetStarted/cmake_options_and_variables.md

```diff
@@ -114,6 +114,32 @@ CMake cannot handle multiple compiler toolchains in one CMake invocation. So the
 above conceptual two-stage compilation happens in two separate CMake
 invocations.
 
+When CMake is invoked with a toolchain file, e.g.,
+[`android.toolchain.cmake`](https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake),
+we get the compiler toolchain for the target platform from the toolchain file.
+IREE under the hood will [create another CMake invocation](https://github.com/google/iree/blob/main/build_tools/cmake/iree_cross_compile.cmake)
+using host platform's compiler toolchain. The inner CMake invocation will be
+placed under `IREE_HOST_BINARY_ROOT`.
+
+The tricky part is to set up target artifacts' dependencies on host executables.
+Normally in CMake we define targets and `target_link_libraries` to express the
+dependency relationship between targets. The optimal way is to [`export`](https://cmake.org/cmake/help/latest/command/export.html)
+targets under host configuration and then [`import`](https://cmake.org/cmake/help/latest/command/add_executable.html?highlight=import#imported-executables)
+into target configuration. But that is not playing well with LLVM's TableGen
+configurations. LLVM's cross-compilation uses on files for dependencies. So
+here we need to do the same. This results in reverting the relationship between
+IREE artifact output names (e.g., `iree-translate`) and IREE package-prefixed
+CMake target names (e.g., `iree_tools_iree-translate`).
```

Comment:
Good details, but a bit worried that they are more developer-oriented (for anyone maintaining IREE's CMake macros themselves).

Reply:
Yup. I feel we need another doc to describe the logic behind build systems. Tracked by https://github.com/google/iree/issues/2111. I'll add a TODO here to move after we have such a doc.